Muzikie/api
# first setup for api project to use web3 solana program

This PR introduces the following changes to enable interaction with the Solana program within the Strapi project:

## Added Solana IDL File:
   The IDL file for the Solana program has been added at ./src/api/project/solana/idl.json.
   The IDL is required to interact with the Solana program using @coral-xyz/anchor.

## Updated Project Service:
   The project.ts service has been extended with a new method interactWithSolanaProgram.
   This method uses the IDL to send instructions to the Solana program and handle transactions.
   The method returns the transaction ID on successful interaction with the Solana blockchain.

## Strapi Integration:
  This updated service can be called from a Strapi controller, allowing the frontend to interact with the Solana program 
   through Strapi’s API.

### Changes Made
 Added: ./src/api/project/solana/idl.json (Solana Program IDL).
 Updated: ./src/api/project/services/project.ts to handle Solana program interaction via the IDL.

### How to Test
 Ensure that you have a valid Solana wallet connected (or simulate one with a local wallet).
 Use Strapi’s API to send a request with the user’s wallet and the appropriate instruction data for the Solana program.
 Verify that a valid transaction ID is returned upon successful execution of the interaction with the Solana program.

### Checklist

- [ ] Added IDL file for the Solana program.
- [ ] Implemented service method for Solana interaction.
- [ ] Tested interaction and transaction flow on Devnet/Mainnet.
